### PR TITLE
feat: add different deterministic sorting of records per user

### PR DESCRIPTION
**Pull request by gabrielmbmb**
# Description

This PR updates the `search` method of the `SearchEngine` to sort the records for each user in a different way using the [`random_score`](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html#function-random) function and the user id as seed for this function to make the sort deterministic between user sessions. In addition, the default sort of the records using the `inserted_at` field in ascending order has been removed.

The affected endpoints by this change are `GET /api/v1/me/datasets/:dataset_id/records` and `GET /api/v1/me/datasets/:dataset_id/records/search`.

## Missing

If we finally decide to have this feature as default in the server, then before merging:

- [ ] Add unit tests using more than one user to check that the sorting of the records is different
- [ ] Add this change in the CHANGELOG.md

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

Change is already being used in https://huggingface.co/spaces/DIBT/prompt-collective

Apart from that, created several users, use them to list the records with the endpoints `GET /api/v1/me/datasets/records` and check that the retrieved records are in different order for all the users.

**Checklist**

- [ ] I added relevant documentation
- [x] I followed the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla_server/search_engine/commons.py
Comment (jfcalvo) [inline src/argilla_server/search_engine/commons.py:250]: Maybe working in positive logic and only using boolean operators is easier to understand:

```suggestion
def is_response_status_scope(scope: FilterScope) -> bool:
    return (
        isinstance(scope, ResponseFilterScope) and
        scope.property == "status" and 
        scope.question is None
    )
```

**Review by jfcalvo — APPROVED** [review]
I added a suggestion in the case that you want to apply it. 

I would wait for @frascuchon approve before merge.

**Review by frascuchon — APPROVED** [review]

**Review by frascuchon — APPROVED** [review]
I changed failing tests to make them work. Great work, @gabrielmbmb!

**Pull request merged**
